### ISSUE TITLE
Revisit compilation flag to prevent adding a single explicitly compiled regex to the cache

### ISSUE THREAD

**Issue by makyen** [mrabarnett/mrab-regex#472]
About a [year and a half ago](https://github.com/mrabarnett/mrab-regex/issues/387), `regex.cache_all()` was added to allow enabling/disabling the implicit regex cache on a global basis. That feature is quite helpful. Unfortunately, we're encountering a situation where that doesn't meet our needs, due to threading.

We have multiple threads scanning through different blocks of text at the same time. For the vast majority of regular expressions, we want the cache to be active, as those regular expressions are invariant, once created. However, we have cases where a regular expression is constructed based on the text which is being tested. It's used briefly within a single thread and never used again. We'd like to be able to explicitly compile such a regular expression without it entering the cache. If it enters the cache, such regexes will cause the cache to continuously fill, resulting in regexes which are static and used thousands of times being expelled from the cache, requiring them to be recompiled.

We could handle this by calling `regex.cache_all(False)` prior to explicitly compiling such a dynamically generated, single use regular expression, but that will turn off caching in all threads, which isn't something we desire. We'd really prefer to turn off caching for just specific calls to `regex.compile`. Overall, the effect would be as if we could call `regex._compile()` and set the `cache_it` argument to `False`.

For us, there's a reasonable workaround we are using. We're currently using:
```
from regex.regex import _compile as regex_raw_compile

def regex_compile_no_cache(regex_text, flags=0, ignore_unused=False, **kwargs):
    return regex_raw_compile(regex_text, flags, ignore_unused, kwargs, False)
```
Then, we use `regex_compile_no_cache()` to compile the regular expressions which we desire to not have in the implicit cache.

**Comment by mrabarnett** [comment]
My preference, I think, would be to add an argument to `compile`:

```
    def compile(pattern, flags=0, ignore_unused=False, cache_pattern=None, **kwargs):
```

That would control whether the pattern is cached: `True` (yes), `False` (no), `None` (current setting of `cache_all`).

**Comment by makyen** [comment]
That seems like a good way to do it.

**Comment by mrabarnett** [comment]
Fixed in regex 2022.6.2.
